Adding playable demo

XmYx opened the pull request:
some performance optimization to infer 'live'

XmYx committed:
Create game.py

XmYx committed:
Update game.py

- fp16
- torch.compile vae
- sliding context window

XmYx committed:
Update rotary_embedding_torch.py

fix tensor.item() access for torch.compile()

Darkblader24 commented:
Damn, this is great! But it's still pretty unplayable, I tested it on a RTX 4090 through WSL and I got like one frame per second.

Danetterry commented:
@Darkblader24, try to set ddim_noise_steps in game.py to lower value like 32 or 16

XmYx committed:
Update game.py

Add fps display (F3)
Adjust steps live (+, -)
Default steps to 16
-----------------------------
4090: 16steps - 3fps

XmYx commented:
> @Darkblader24, try to set ddim_noise_steps in game.py to lower value like 32 or 16

I did! Added control to do so while playing too, and a simple fps counter. 10 steps: 6fps, 16steps: 3fps
We should get above 10 easily when compiled, and potentially an other sampler.

XmYx committed:
Update dit.py

less einops

XmYx committed:
Update game.py

enable torch compile model - 9fps
F4 to reset to start frame

XmYx commented:
Currently at 9fps at 16 steps compiled.

Darkblader24 commented:
I got it working on Windows and Python 3.12 by installing Triton for Windows with this command:
`pip install https://huggingface.co/madbuda/triton-windows-builds/resolve/main/triton-3.0.0-cp312-cp312-win_amd64.whl`

With a RTX 4090 on Windows I get 2 FPS, but running it on WSL I get 7 FPS. Seems like running it on Linux natively is much better. But on Linux via WSL it fails to capture my mouse, so this is rather annoying.

XmYx committed:
Create train.py

add train script trial

XmYx commented:
Added a trial for a train.py

XmYx committed:
Update train.py

actually use the whole video

XmYx committed:
Merge branch 'etched-ai:master' into master

XmYx committed:
Merge branch 'etched-ai:master' into master

XmYx committed:
Update game.py

suppress torch compile errors

sal34 commented:
performance on a **3060 12GB** Win 11, with triton for win python 3.11 + compiled torch using 16 steps, I got **1fps**.

XmYx commented:
wondering, @julian-q if this train.py is anything close to the original : )

julian-q commented:
Updated way context noising is done -- try to merge with the latest changes and use 10 DDIM steps. You should hopefully see a speedup!

julian-q commented:
I would love to include this pygame interactive functionality, so thank you for contributing! Can you try to incorporate the pygame functionality into the original `generate.py` script to avoid code duplication?

Maybe by decomposing the pygame stuff into some sort of `pygame_utils.py` or something. Let me know what you think!

XmYx committed:
Update game.py

add stabilization

XmYx committed:
Update rotary_embedding_torch.py

XmYx committed:
Merge branch 'master' into master

XmYx commented:
Absolutely, currently was running into issues with the generate, but updated the pygame with the stability aiming changes, and resolved conflicts for merging.

julian-q commented:
Great - I'm still seeing a lot of code duplicated across `generate.py` and `game.py` though. Want to try adding the pygame functionality to the latest `generate.py` script, remove `game.py`, and include any extra functions you need in `utils.py`?

Also for the scope of a single PR let's remove `train.py`. Though love to see you experimenting with training!
